src/gui/doAction.cpp

```diff
@@ -883,6 +883,38 @@ void FurnaceGUI::doAction(int what) {
     case GUI_ACTION_WAVE_LIST_SAVE_RAW:
       if (curWave>=0 && curWave<(int)e->song.wave.size()) openFileDialog(GUI_FILE_WAVE_SAVE_RAW);
       break;
+    case GUI_ACTION_WAVE_LIST_CREATE_SAMPLE:
+      if (curWave>=0 && curWave<(int)e->song.wave.size()) {
+        curSample=e->addSample();
+        if (curSample==-1) {
+          showError(_("too many samples!"));
+        } else {
+          e->lockEngine([this]() {
+            DivSample* sample=e->getSample(curSample);
+            if (sample!=NULL) {
+              DivWavetable* wave=e->song.wave[curWave];
+              unsigned int waveLen=wave->len;
+              sample->rate=(int)round(261.625565301*waveLen); // c3
+              sample->centerRate=(int)round(261.625565301*waveLen); // c3
+              sample->loopStart=0;
+              sample->loopEnd=waveLen;
+              sample->loop=true;
+              sample->loopMode=DIV_SAMPLE_LOOP_FORWARD;
+              sample->depth=DIV_SAMPLE_DEPTH_8BIT;
+              if (sample->init(waveLen)) {
+                for (unsigned short i=0; i<waveLen; i++) {
+                  sample->data8[i]=wave->data[i]-waveLen/2;
```

Comment:
I have a feeling there is something wrong in this line. I will test later.

Reply:
oh right i cant center using wave length
i had to use wave height